api/bootstrap/v1beta1/k0s_types.go

```diff
@@ -64,6 +64,11 @@ type K0sWorkerConfigSpec struct {
 	// +kubebuilder:validation:Optional
 	Version string `json:"version,omitempty"`
 
+	// UseSystemHostname specifies whether to use the system hostname for the kubernetes node name.
+	// By default, k0s will uses Machine name as a node name. If true, it will pick it from `hostname` command output.
+	// +kubebuilder:validation:Optional
```

Comment:
Just to be explicit, should we define `false` as the default value?

Reply:
Sorry, forgot about this. Fixed!